Added yet more destructuring tests.

webbedspace opened the pull request:
Also, reordered the compilers in build.js, and removed a console.log() from master.js.

webbedspace committed:
Added yet more destructuring tests.

Also, reordered the compilers in build.js, and removed a console.log() from master.js.

kangax commented:
@webbedspace out of curiosity, which criteria did you reorder compilers by?

kangax commented:
@webbedspace inspired by Axel's latest by any chance? :) Been meaning to read it — looked great — but didn't get a chance yet.

kangax commented on data-es6.js:
```diff
@@ -3715,6 +3739,44 @@ exports.tests = [
+    },
+    'throws on null and undefined': {
+      exec: function(){/*
+        var n = null;
```
Hm, why store them as variables instead of just using inline?

webbedspace replied:
One of the compilers (can't remember which) throws a compile error when it encounters a null literal as a destructuring source, which is technically a spec violation (it should throw a TypeError at runtime), but behaves on-spec if it's in a variable.

kangax replied:
Interesting. I feel like we probably shouldn't be "masking" it then. But it's certainly tricky if it completely throws compiler off not letting even get output file...

webbedspace replied:
Actually, nevermind - it turns out the compile error was for some other aspect of the test's structure (namely, that `var {a}` appeared twice, which for some reason it didn't like) which has since been rewritten.

webbedspace commented:
Compilers are reordered in terms of how long they take to build their pages, with Closure currently the slowest.

webbedspace committed:
Updated WebKit results, and a few others.

Also reworded certain destructuring tests so that they'll provide the most accurate results in some particularly buggy environments. For instance, destructuring assignments outside of var statements are now checked as basic functionality.

Also tweaked ES6-Transpiler's test page output so that error messages are properly printed.

webbedspace committed:
Slightly tweaked the 'destructuring null or undefined' test.

kangax commented:
Thanks!
